src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    const currentAnimal = Animal.alive.find(animal => animal === herbivore);
+
+    if (currentAnimal.hidden !== undefined && currentAnimal.hidden !== true) {
+      currentAnimal.health -= 50;
+    }
+
+    if (currentAnimal.health <= 0) {
+      Animal.alive = Animal.alive.filter(animal => animal !== currentAnimal);
+    }
+  }
```

Comment:
Not sure why would you create `currentAnimal`. `herbivore` is already the current animal, there is no need for extra variable. Everything should work with simply using the `herbivore` parameter instead of every instance of `currentAnimal`
```suggestion
    if (herbivore.hidden !== undefined && herbivore.hidden !== true) {
      herbivore.health -= 50;
    }

    if (herbivore.health <= 0) {
      Animal.alive = Animal.alive.filter(animal => animal !== herbivore);
    }
  }
```

Reply:
Agree with @glugovsky

Reply:
yeah, I see that there was no reason to do it, fixed, thanks)